### DISCUSSION TITLE
How can i initialize agent variables conditioning to environment properties

### DISCUSSION THREAD

**Discussion by khateraAlizada** [FLAMEGPU/FLAMEGPU2#1023] [Q&A]
How can I access environment Property in function def initialise_simulation
.......
I initialize agents variables based on value of an environment variable "probability-i=1"
# Initialize nation states
      nationStatePopulation = pyflamegpu.AgentVector(model.Agent("nation-state"), num_nationState)
      for i in range(0, num_nationState):
        ns = nationStatePopulation[i]
     
 p =  pyflamegpu.environment.getPropertyFloat("probability-i=1")  //This line gives error "AttributeError: module 'pyflamegpu' has no // attribute 'environment'"
        r = pyflamegpu.random.uniform(0, 1)
        if r < p:
          ns.setVariableInt("has-i", 1)
        else:
          ns.setVariableInt("has-i", 0)

**Comment by Robadob** [comment]
I would suggest you instead create the agents within a initialisation function. This gives you access to the `HostAPI`, which has an `environment` attribute.

https://docs.flamegpu.com/guide/host-functions/index.html

If you are creating your agents outside of the model (e.g. before you call `simulate()`), you will need to store your own copies of the environment properties initial values, or retrieve them with `getEnvironmentProperty()` from the `CUDASimulation` object you have created ([C++ API Docs](https://docs.flamegpu.com/guide/host-functions/index.html), I don't think this function is covered in the userguide, but it should work in Python via the equivalents e.g. `getEnvironmentPropertyFloat()`, `getEnvironmentPropertyArrayInt()` etc).
